From a263157165c411552d73a211ac9c7647285abdc5 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Sun, 10 Jun 2018 22:01:47 +0300
Subject: [PATCH] Fix build without SSL support

The <memory> headers is needed for unique_ptr even when SSL is not
enabled. This fixes the following build failure:

src/Utils.cpp: In static member function 'static bool CUtils::CheckCIDR(const CString&, const CString&)':
src/Utils.cpp:674:5: error: 'unique_ptr' is not a member of 'std'
     std::unique_ptr<addrinfo, decltype(deleter)> aiHost(aiHostC, deleter);
     ^

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: https://github.com/znc/znc/pull/1554

 src/Utils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Utils.cpp b/src/Utils.cpp
index 9c3b2549b791..cced5683d78a 100644
--- a/src/Utils.cpp
+++ b/src/Utils.cpp
@@ -27,8 +27,8 @@
 #include <znc/Message.h>
 #ifdef HAVE_LIBSSL
 #include <openssl/ssl.h>
-#include <memory>
 #endif /* HAVE_LIBSSL */
+#include <memory>
 #include <unistd.h>
 #include <time.h>
 
-- 
2.17.1

